packages/syft/src/syft/grid/client/request_api/dataset_api.py

```diff
@@ -65,14 +65,10 @@ def to_obj(self, result: Any) -> Any:
         dataset_obj.files = datasets
         type(dataset_obj).__getitem__ = lambda x, i: x.data[i]
         dataset_obj.client = self.client
-        return Dataset(dataset_obj)
-
-    def _repr_html_(self) -> str:
-        """Displays a nice table when the object is returned in Jupyter Notebook"""
-        return self.all(pandas=True)._repr_html_()  # type: ignore
+        return DatasetPointer(dataset_obj)
 
 
-class Dataset:
+class DatasetPointer:
```

Comment:
Why can't we reuse the Pointer generation API? I would like to reuse or pointer creation mechanism so we rely on existing working code.

Reply:
Would love to if we can, although I do have lots of custom UX things that the pointer generation API probably won't get us (like printing things as numpy tables etc.)